### ISSUE TITLE
Command-line argument error

### ISSUE THREAD

**Issue by bentsherman** [SystemsGenetics/KINC#18]
When I try to run KINC with ACE console, I get an argument error with both clustering analytics. To make sure it wasn't just me, I also tried to run spearman and got the same error:

```
$ build/KINC run spearman --input "data/Yeast.emx" --output "data/Yeast.cmx" --min 30 --minthresh 0.5 --maxthresh 1 --bsize 4 --ksize 4096
build/KINC: /usr/local/cuda/lib64/libOpenCL.so.1: no version information available (required by build/KINC)
build/KINC: /usr/local/cuda/lib64/libOpenCL.so.1: no version information available (required by /home/bent/software/lib/libacecore.so.0)
0%CRITICAL ERROR

Argument Error

Did not get valid input and/or output arguments.

File: ../src/spearman.cpp
Line: 237
Function: virtual bool Spearman::initialize()
```

The command was taken verbatim from the KINC GUI. It looks like input / output data objects aren't being initialized properly, even though the ACE source code looks good and this operation works with ACE GUI. As a result I haven't been able to run anything from the command-line yet.

(Also, the version information lines don't seem to cause any problems but if you know how to get rid of them that would be nice to know.)

**Comment by 4ctrl-alt-del** [comment]
Thanks for finding that bug I will work on resolving right away.

**Comment by bentsherman** [comment]
Currently doing some of my own tracing. In `Spearman::setArgument(int argument, EAbstractData *data)` the `data` argument is valid in both cases but `_output` is null on exit whereas `_input` is fine. So I'm guessing the `dynamic_cast` is failing because `data` is not a `CorrelationMatrix*`.

**Comment by bentsherman** [comment]
I think I found the bug, in `console/Application.cpp` in `EApplication::run()`:
```
      case Type::DataIn:
         analytic->addDataIn(*j,value,analytic->getArgumentData(i,Role::DataType).toUInt());
         break;
      case Type::DataOut:
         analytic->addDataOut(*j,value,analytic->getArgumentData(i,Role::DataType).toUInt());
         break;
```

Each instance of `i` should be replaced with `*j`. The `DataIn` option works because `i` happens to equal `*j` for the first argument.

Anyway, I can work with this quick fix for now on my machine, but Josh I'll leave it to you to examine this fix and push it as you see fit.

**Comment by 4ctrl-alt-del** [comment]
I can confirm that was a bug. Thanks for tracking that down and finding it!

I will patch ACE right away and a fix to this bug will be part of the 2.1 release.

**Comment by feltus** [comment]
Can Qt be put in a container?


Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Josh B <notifications@github.com> Date: 1/11/18  5:05 PM  (GMT-05:00) To: SystemsGenetics/KINC <KINC@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [SystemsGenetics/KINC] Command-line argument error (#18) 
I can confirm that was a bug. Thanks for tracking that down and finding it!
I will patch ACE right away and a fix to this bug will be part of the 2.1 release.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/SystemsGenetics/KINC","title":"SystemsGenetics/KINC","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/SystemsGenetics/KINC"}},"updates":{"snippets":[{"icon":"PERSON","message":"@4ctrl-alt-del in #18: I can confirm that was a bug. Thanks for tracking that down and finding it!\r\n\r\nI will patch ACE right away and a fix to this bug will be part of the 2.1 release."}],"action":{"name":"View Issue","url":"https://github.com/SystemsGenetics/KINC/issues/18#issuecomment-357076615"}}}

**Comment by bentsherman** [comment]
@feltus Yes, so the issue of building ACE for console / gui won't matter for any of our containerized applications.
